### PR TITLE
fixed movetoworkspace and movetoworkspacesilent 

### PR DESCRIPTION
**Pull request by levnikmyskin**
Hi!
the `movetoworkspace` and `movetoworkspacesilent` were not properly working with special workspaces. This should fix it now.   
  
Please notice I didn't check if there were other broken dispatchers.

Cheers :)

### CODE REVIEW

File: src/dispatch.rs [hunk 1]
```diff
@@ -467,9 +473,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
         KillActiveWindow => "killactive".to_string(),
         CloseWindow(win) => format!("closewindow{sep}{win}"),
         Workspace(work) => format!("workspace{sep}{work}"),
-        MoveToWorkspace(work, Some(win)) => format!("movetoworkspace {work} {win}"),
+        MoveToWorkspace(work, Some(win)) => format!("movetoworkspace {work},{win}"),
         MoveToWorkspace(work, None) => format!("movetoworkspace {work}"),
-        MoveToWorkspaceSilent(work, Some(win)) => format!("movetoworkspacesilent {work} {win}"),
+        MoveToWorkspaceSilent(work, Some(win)) => format!("movetoworkspacesilent {work},{win}"),
```
Comment (yavko) [inline src/dispatch.rs:478]: Could you add `{sep}` so it also works in binds?

File: src/dispatch.rs [hunk 2]
```diff
@@ -467,9 +473,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
         KillActiveWindow => "killactive".to_string(),
         CloseWindow(win) => format!("closewindow{sep}{win}"),
         Workspace(work) => format!("workspace{sep}{work}"),
-        MoveToWorkspace(work, Some(win)) => format!("movetoworkspace {work} {win}"),
+        MoveToWorkspace(work, Some(win)) => format!("movetoworkspace {work},{win}"),
```
Comment (yavko) [inline src/dispatch.rs:476]: Could you add `{sep}` so it also works in binds?

File: src/dispatch.rs [hunk 3]
```diff
@@ -467,9 +473,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
...
         MoveToWorkspace(work, None) => format!("movetoworkspace {work}"),
```
Comment (yavko) [inline src/dispatch.rs:477]: Could you add `{sep}` so it also works in binds?

File: src/dispatch.rs [hunk 4]
```diff
@@ -467,9 +473,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
...
-        MoveToWorkspaceSilent(work, Some(win)) => format!("movetoworkspacesilent {work} {win}"),
+        MoveToWorkspaceSilent(work, Some(win)) => format!("movetoworkspacesilent {work},{win}"),
         MoveToWorkspaceSilent(work, None) => format!("movetoworkspacesilent {work}"),
```
Comment (yavko) [inline src/dispatch.rs:479]: Could you add `{sep}` so it also works in binds?

**Review by yavko — CHANGES_REQUESTED** [review]
Looks good, just could you add support for binds? This function is also used in `config::binds::Binder::gen_str` :)

**Comment by levnikmyskin** [comment]
Yeah sure, but I don't know how `sep` works here. From line 467:
`let sep = if dispatch { " " } else { "," };`

Does this mean `sep` would be a white space here? because if so, then this wouldn't work.

**Comment by levnikmyskin** [comment]
Hey, sorry for the delay. Been pretty busy. Should be fine now, right?

**Review by yavko — APPROVED** [review]
lgtm

**Pull request merged**